Introduce a new package for containerd and related binaries

vikramhh opened the pull request:
1. Restructure code to make it easier to have multiple packages.
2. Introduce the containerd package which will make it possible
    to service containerd/processshim and related binaries 
    independently of Docker EE binaries.
3. Introduce implicit links between a package and its source's
   names (a source for a given package should start with that
   package's name)
4. Handle multiple names and wildcards in package name
   specification
5. Installation of the containerd service.
6. Removal of redundant/duplicated code and bug fixes for
   issues encountered while making changes.

Signed-off-by: Vikram bir Singh <vikrambir.singh@docker.com>

Commit:
Introduce a new package named containerd

1. Restructure code to make it easier to have multiple packages.
2. Introduce the containerd package
3. Introduce implicit links between a package and its source's
   names (a source for a given package should start with that
   package's name)
4. Handle multiple names and wildcards in package name
   specification
5. Installation of the containerd service.
6. Removal of redundant/duplicated code and bug fixes for
   issues encountered while making changes.

Pending:

1. Specification and handling of Dependencies - future work.

Signed-off-by: Vikram bir Singh <vikrambir.singh@docker.com>

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/signed)](https://cla.opensource.microsoft.com/OneGet/MicrosoftDockerProvider?pullRequest=66) <br/>All CLA requirements met.

vikramhh commented:
WIP: It is intended to update the with new dependency logic so `Install-Package Docker -ProviderName DockerMsftProvider -Force` will automatically install all dependencies. It is expected that most consumers will go this route.

vikramhh commented:
We need the Containerd metadata hosting URL from Microsoft [similar to what we have for Docker] - https://github.com/OneGet/MicrosoftDockerProvider/pull/66/files#diff-8c83ac27906d37a3491f485daf8c5781R29

For now a placeholder has been used - it is good enough for testing.

Commit:
Docker default install should install containerd

The default installation of Docker [without specifying
a version number] should lead to installation of the
containerd package as well.

Signed-off-by: vikrambirsingh <vikrambir.singh@docker.com>

vikramhh commented:
Updated the PR by adding another commit which will install containerd package as well when the "default" install of Docker package is requested [i.e. install of Docker package without specifying any version number etc. - this is the way most consumers invoke the install].
